rocketpy/rocket/rocket.py

```diff
@@ -1243,7 +1254,11 @@ def add_air_brakes(
             name=name,
         )
         controller = Controller(
-            air_brakes, controller_function, sampling_rate, controller_name
+            interactable_objects=air_brakes,
+            controller_function=controller_function,
+            sampling_rate=sampling_rate,
+            initial_observed_variables=initial_observed_variables,
+            name=controller_name,
         )
         self.air_brakes.append(air_brakes)
```

Comment:
don't we have a self.add_air_brakes(air_brakes) method? These two rows (1263 and 1264) does the same thing but in two different ways.

Reply:
I don't think I understood what you were saying here

Reply:
you have used these two code lines:

`self.air_brakes.append(air_brakes)`

and

`self.add_controllers(controller)`

My question is: why do you append airbakes but you add controllers? Shouldn't they follow the same architecture? The code could be more intuitive

Reply:
`add_controllers` is a separate method that can be used by itself to add any type of controller to the rocket

the `add_air_brakes` is specific for adding air_brakes. It also adds the controller because it just makes sense for air brakes to be added together with their controller

Reply:
> `add_controllers` is a separate method that can be used by itself to add any type of controller to the rocket
> 
> the `add_air_brakes` is specific for adding air_brakes. It also adds the controller because it just makes sense for air brakes to be added together with their controller

Now I'm the one who thinks you didn't understand...

The point is that in your code you didn't use `add_air_brakes` at all, instead you are mutating a list using the `self.airbrakes.append()`.

Imagine when we are on the Rocket class, we don't do this: `calisto.aero_surfaces.append(...)`.
Instead, we do `calisto.add_surface(...)`.
This is important because there are some function calls inside the add_surface method that make the first approach not valid.

Reply:
`add_air_brakes` creates the air brakes and their related `controller` and saves them in their appropriate lists.

Air brakes do not generate any lift, so there is no reason for it to be included in `aerodynamic_surfaces` or make any center of pressure-related calculations. So it does not call `add_surface`

Reply:
Also, air brakes have no position, so it only needs to be saved in a normal list. No need to create a `Component` object